### PR TITLE
Add aggregation for specific resource listings

### LINKED ISSUE #4056: Add aggregations options to specific resource listings
**Motivation**
In #3944 , aggregations were introduced to get statistics about the number of projects and types for all resources.
This will allow to extend it to the listing operations for files / storages / resolvers and the different views

**Acceptance criteria**
1. When adding the `aggregations=true` parameter on listings for files/storages/resolvers/views
2. Then aggregations for projects and types is returned in the same way as for the generic resources endpoint, only being further restricted to the types linked by this given endpoint

### PR DESCRIPTION
**Pull request by olivergrabinski**
Closes #4056

### CODE REVIEW

File: delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/routes/ElasticSearchViewsDirectives.scala [hunk 1]
```diff
@@ -45,8 +45,11 @@ trait ElasticSearchViewsDirectives extends UriDirectives {
   /**
     * Matches only if the ''aggregations'' parameter is set to ''true''
     */
-  def aggregated: Directive1[Option[Boolean]] =
-    parameter("aggregations".as[Boolean].?).filter(_.contains(true))
+  def aggregated: Directive0 =
+    parameter("aggregations".as[Boolean].?).flatMap {
+      case Some(true) => pass
+      case _          => reject
+    }
```
[the remark this answers is not in the document]
Comment (shinyhappydan) [reply delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/routes/ElasticSearchViewsDirectives.scala:52]: This feature brought to you from @dantb

File: docs/src/main/paradox/docs/delta/api/schemas-api.md [hunk 2]
```diff
@@ -334,7 +337,7 @@ GET /v1/schemas?from={from}
   (containing) the provided string
 - `{sort}`: String - can be used to sort schemas based on a payloads' field. This parameter can appear multiple times 
   to enable sorting by multiple fields. The default is done by `_createdBy` and `@id`.
-
+- `{aggregations}`: Boolean - if `true` then the response will only contain aggregations of the `@type` and `_project` fields; defaults to `false`. See @ref:[Aggregations](resources-api.md#aggregations).
```
Comment (shinyhappydan) [inline docs/src/main/paradox/docs/delta/api/schemas-api.md:340]: It's possible that `aggregate` makes more sense since it's a command?

**Review by imsdu — DISMISSED** [review]

File: tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/AggregationsSpec.scala [hunk 3]
```diff
@@ -76,6 +78,54 @@ final class AggregationsSpec extends BaseSpec with Inspectors with EitherValuabl
...
+      val expected = jsonContentOf(
+        "/kg/aggregations/resolvers-aggregation.json",
+        "org"     -> org1,
+        "project" -> proj11
+      )
+      deltaClient.get[Json](s"/resolvers/$ref11?aggregations=true", Charlie) { (json, response) =>
+        response.status shouldEqual StatusCodes.OK
+        json should equalIgnoreArrayOrder(expected)
+      }
+    }
```
Comment (shinyhappydan) [inline tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/AggregationsSpec.scala:91]: I would love if this test were expressing the intent more

**Review by shinyhappydan — DISMISSED** [review]

File: docs/src/main/paradox/docs/delta/api/schemas-api.md (hunk 2 above)
Comment (olivergrabinski) [reply docs/src/main/paradox/docs/delta/api/schemas-api.md:340]: We already had this discussion when we introduced aggregation on generic resources and this was the result. One of the reasons was there are already aggregate views around which are a different thing

**Review by shinyhappydan — APPROVED** [review]

**Pull request merged**